op-node/rollup/derive/channel_bank.go

```diff
@@ -38,18 +38,20 @@ type ChannelBank struct {
 	progress Progress
 
 	next ChannelBankOutput
+	prev *L1Retrieval
 }
 
 var _ Stage = (*ChannelBank)(nil)
 
 // NewChannelBank creates a ChannelBank, which should be Reset(origin) before use.
-func NewChannelBank(log log.Logger, cfg *rollup.Config, next ChannelBankOutput) *ChannelBank {
+func NewChannelBank(log log.Logger, cfg *rollup.Config, next ChannelBankOutput, prev *L1Retrieval) *ChannelBank {
```

Comment:
Will focus on the retrieval code for this PR, the channel bank can change later. But would be nice to separate each stage by interface instead of concrete types.

Reply:
fyi the channel bank takes an interface rather than a concrete type in the next PR